### PR TITLE
PHP 8.0 Compatibility

### PR DESCRIPTION
**Pull request by ausi**
As discussed in #487

This is currently a draft. ☺️

### CODE REVIEW

**Comment by ausi** [comment]
All dependencies except `php-http/guzzle6-adapter` are compatible with PHP 8. I’m not sure if we can get rid of `php-http/guzzle6-adapter` easily? But because it is only a dev dependency it should not hold us back from marking this package compatible with php 8 (if everything else works).

**Comment by dbu** [comment]
thanks! 

i suggest to use guzzle7-adpater, as guzzle 6 is not compatible with php 8. guzzle 7 works with php 7.2 - 8.0.
lets drop php 7.1 support to keep things simple. (and users on php 7.1 can keep using the current version, its not like we have a lot of new features happening at the moment)

i fixed the styleci error in a separate branch, please rebase.
can you please put the branch alias in composer.json to 2.10.x? (it should currently be 2.9.x but was forgotten to be updated). and please add a note in the CHANGELOG.md file

**Comment by ausi** [comment]
I was unable to fix the `sphinxcontrib-spelling requires Python '>=3.6' but the running Python is 2.7.6` error in
<https://travis-ci.com/github/FriendsOfSymfony/FOSHttpCache/jobs/452324683#L856>

But I think it is not related to this pull request

File: .travis.yml [hunk 1]
```diff
@@ -10,21 +10,25 @@ env:
...
+    - php: 7.2
       env: VARNISH_VERSION=3.0 SYMFONY_PHPUNIT_VERSION=5.7 COMPOSER_FLAGS="--prefer-lowest" DEPENDENCIES=""
 
-    - php: 7.1
-      env: SYMFONY_PHPUNIT_VERSION=7
     - php: 7.2
     - php: 7.3
     - php: 7.4
       env: DOCCHECK=true COVERAGE=true PHPUNIT_FLAGS="-v --coverage-text" VARNISH_MODULES_VERSION=0.12.1
+      python: 3.6
```
Comment (dbu) [inline .travis.yml:26]: lets drop this line as it did not help. we can ignore the failure and i will comment out the doccheck in a separate PR

**Review by dbu — CHANGES_REQUESTED** [review]
awesome, thanks a lot! ok to remove the python version again? we can try to fix that in a different PR

File: .travis.yml (hunk 1 above)
Comment (ausi) [reply .travis.yml]: Done in a05cb960adf73894e936e746413eb6e890b69cf9

**Comment by ausi** [comment]
Everything else is green now 🎉

**Review by dbu — APPROVED** [review]
thanks a lot!

i will tag the release after merging this.

i skip the doccheck for now, if nobody fixed it in the last 1.5 years, chances are it won't soon be fixed anyways...

**Comment by ausi** [comment]
Thank you @dbu

Should I create a pull request for https://github.com/FriendsOfSymfony/FOSHttpCacheBundle too?

**Comment by dbu** [comment]
if you have the time to do it, i'd appreciate it :+1:

**Comment by ausi** [comment]
I will look into it.

**Pull request merged**